### PR TITLE
[flux-core v0.13] example 7 update: Update bookkeeper script, README, blacken Python scripts

### PR DESCRIPTION
**Pull request by cmoussa1**
This PR proposes the following changes:

1. **README.md**
- Update format of instructions from bullet points to numbered instructions
- Remove instruction to set `FLUX_SCHED_OPTIONS` environment variable
- Update example output to match new **bookkeeper.py** script

2. **bookkeeper.py**
- Update job submissions in script to comply with new Python bindings, create new callback function that waits for jobs to transition to `INACTIVE` state

3. **compute.py** and **io-forwarding.py**
- Blacken both scripts

### CODE REVIEW

**Comment by cmoussa1** [comment]
Added another commit that affects **README.md**:

- added more verbosity to the instructions contained in this example that goes into more detail about interacting with Flux. 
- added a **Notes** section that describes some of the highlights in the source code contained in the example. The descriptions are just my version and probably a rough draft; if anybody has a better suggestion for how to word those descriptions feel free to let me know!

File: example7/compute.py [hunk 1]
```diff
@@ -3,12 +3,15 @@
 import argparse
 import time
 
-parser = argparse.ArgumentParser (description='compute for seconds')
-parser.add_argument ('integer', metavar='S', type=int,
-                     help='an integer for the number of seconds to compute')
+parser = argparse.ArgumentParser(description="compute for seconds")
```
Comment (dongahn) [inline example7/compute.py]: interesting. How did you apply black formatting to this?

File: example7/bookkeeper.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def submit_bundles(f, N):
         print (flux.job.submit(f, compute_jobreq))
         print (flux.job.submit(f, io_jobreq))
 
-    print "bookkeeper: all jobs submited"
+    print "bookkeeper: all jobs submitted"
```
Comment (dongahn) [inline example7/bookkeeper.py]: Please see https://github.com/flux-framework/flux-sched/pull/565

You will find the comment about the synax changes of print with python 3. You might as well make that change. Thanks.

File: example7/README.md [hunk 3]
```diff
@@ -32,3 +38,31 @@ job 285564993536 changed its state to INACTIVE
...
+
+---
+
+##### Notes
+
+- `f = flux.Flux()` creates a new Flux handle which can be used to connect to and interact with a Flux instance.
+
+
+- The following constructs a job request using the **JobspecV1** class with customizable parameters for how you want to utilize the resources allocated for your job:
+```
```
Comment (dongahn) [inline example7/README.md]: \`\`\`python \`\`\` will add color to the enclosing Python code.

File: example7/README.md [hunk 4]
```diff
@@ -32,3 +38,31 @@ job 285564993536 changed its state to INACTIVE
...
+    command=["./compute.py", "120"], num_tasks=4, num_nodes=2, cores_per_task=2
+)
+compute_jobreq.cwd = os.getcwd()
+compute_jobreq.environment = dict(os.environ)
+```
+
+- `flux.job.submit(f, compute_jobreq)` submits the job to be run, and returns a job ID once it begins running.
+
+- Throughout the course of a job, its state will go through a number of changes. The following subscribes to the event messages matching the transition of those states in the jobs submitted.
+```
```
Comment (dongahn) [inline example7/README.md]: \`\`\`python \`\`\` will add color to the enclosing Python code.

File: example7/compute.py (hunk 1 above)
Comment (cmoussa1) [reply example7/compute.py]: Just with `black compute.py`. Sorry, am I missing something else I should have done?

File: example7/bookkeeper.py (hunk 2 above)
Comment (cmoussa1) [reply example7/bookkeeper.py]: Ah, good catch - I will fix this.

File: example7/compute.py (hunk 1 above)
Comment (dongahn) [reply example7/compute.py]: No. This was for my own edification.

**Comment by cmoussa1** [comment]
OK, rebased and force-pushed some fixes! Thanks @dongahn for your initial feedback - I think this might be ready for another review.

**Comment by dongahn** [comment]
Great. LGTM. Merging.

**Comment by cmoussa1** [comment]
Thanks @dongahn !

**Pull request merged**